### PR TITLE
Pass the StatsCollector into the core Aura metrics classes.

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

File: aerospike/src/main/java/net/opentsdb/aura/metrics/ASBatchClient.java [hunk 1]
```diff
@@ -111,28 +132,28 @@
   private volatile long bytesRead;
   private volatile long bytesDiscarded;
 
-  private volatile Timer callTimer;
-  private volatile Timer overallTimer;
-  private volatile Timer timeToFirstByteTimer;
-  private Timer nodesPerBatch; // not really a timer
-  private Timer keysPerNodePerBatch; // not really a timer
-  private Timer hitsPerNode; // not really a timer
+//  private volatile Timer callTimer;
```
Comment (anonthing) [inline aerospike/src/main/java/net/opentsdb/aura/metrics/ASBatchClient.java:135]: delete me please.

File: aws/src/main/java/net/opentsdb/aura/aws/S3Uploader.java [hunk 2]
```diff
@@ -31,9 +29,19 @@
 import software.amazon.awssdk.services.s3.model.PutObjectResponse;
 
 import java.io.ByteArrayInputStream;
+import java.time.temporal.ChronoUnit;
 
 public class S3Uploader implements Uploader {
```
Comment (anonthing) [inline aws/src/main/java/net/opentsdb/aura/aws/S3Uploader.java:34]: should this class be named as `MetaS3Uploader`?

File: core/src/main/java/net/opentsdb/aura/metrics/core/TSFlusherImp.java [hunk 3]
```diff
@@ -193,17 +199,16 @@ public FlushStatus flushShard(
...
     private final int shardId;
     private final LongLongIterator iterator;
     private final LongLongHashTable timeSeriesTable;
     private final int baseTimestamp;
     private final Job[] jobs;
-    private final Gauge flushedGuage;
-    private final Gauge oooGuage;
-    private final Gauge failureGuage;
-    private final Gauge emptyGuage;
+//    private final Gauge flushedGuage;
```
Comment (anonthing) [inline core/src/main/java/net/opentsdb/aura/metrics/core/TSFlusherImp.java:208]: delete me please.

File: core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java [hunk 4]
```diff
@@ -451,62 +450,52 @@ void testMarkFlushed() {
...
-        assertEquals(3, segmentCount);
-      }
-    };
+//    new Verifications() {
+//      {
+//        long segmentCount;
+//        segmentCountGauge.set(segmentCount = withCapture());
+//        assertEquals(3, segmentCount);
+//      }
+//    };
```
Comment (anonthing) [inline core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java:479]: delete

File: core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java [hunk 5]
```diff
@@ -451,62 +450,52 @@ void testMarkFlushed() {
...
-        assertEquals(segmentCount, 2);
-      }
-    };
+//    new Verifications() {
+//      {
+//        long segmentCount;
+//        segmentCountGauge.set(segmentCount = withCapture());
+//        assertEquals(segmentCount, 2);
+//      }
+//    };
```
Comment (anonthing) [inline core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java:491]: more delete :)

**Review by anonthing — CHANGES_REQUESTED** [review]
just nits.

File: aws/src/main/java/net/opentsdb/aura/aws/S3Uploader.java (hunk 2 above)
Comment (manolama) [reply aws/src/main/java/net/opentsdb/aura/aws/S3Uploader.java:34]: Probably eventually

File: core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java (hunk 4 above)
Comment (manolama) [reply core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java:479]: Keepin these for now to remind myself to fix em

File: core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java (hunk 5 above)
Comment (manolama) [reply core/src/test/java/net/opentsdb/aura/metrics/core/gorilla/GorillaRawTimeSeriesEncoderTest.java:491]: ditto

**Review by anonthing — APPROVED** [review]

**Pull request merged**